src/modals/examples/modal-overlay.js

```diff
@@ -0,0 +1,203 @@
+    <button ng-click="open()" class="btn btn-default">Launch Modal Overlay</button>
+    <pf-modal-overlay show-modal="showModal"
+          on-close="onClose(dismissCause)"
+          modal-id="modalId"
+          modal-title="modalTitle"
+          background-close="backgroundClose"
+          on-background-click="onBackgroundClick"
+          is-form="isForm"
+          modal-body-template="template"
+          modal-body-scope="formScope"
+          action-buttons="actionButtons">
+    </pf-modal-overlay>
+
+   <button ng-click="open2()" class="btn btn-default">Launch Second Modal Overlay</button>
+   <pf-modal-overlay show-modal="showModal2"
+         on-close="onClose2(dismissCause)"
+         modal-id="modalId2"
+         modal-title="modalTitle2"
+         title-id="titleId2"
+         hide-close-icon="hideCloseIcon2"
+         modal-body-template="template2"
+         action-buttons="actionButtons2">
+    </pf-modal-overlay>
+   <div>
+      <label class="actions-label">Actions: </label>
+   </div>
+   <div>
+      <textarea rows="3" class="col-md-12">{{actionsText}}</textarea>
+   </div>
+ </div>
+ </file>
+
+ <file name="script.js">
+ angular.module('patternfly.modals').controller('DemoModalOverlayCtrl', function( $scope ) {
+
+      $scope.actionsText = "";
+
+      // first example
+      $scope.open = function () {
+          $scope.showModal = true;
+       };
+      $scope.onClose = function(dismissCause) {
+          $scope.showModal = false;
+          $scope.actionsText = "First Modal closed via: " + dismissCause + "\n" + $scope.actionsText;
+       };
+
+      $scope.modalId = "demoModal1";
+      $scope.modalTitle = "First Demo Title";
+      $scope.backgroundClose = true;
+      $scope.onBackgroundClick = function() {
+        if(!$scope.formScope.allowBackgroundDismissal) {
+          $scope.actionsText = "Backdrop clicked, but dismissal using background is not allowed\n" + $scope.actionsText;
+          $scope.formScope.showNotAllowedMsg = true;
+          return false;
+        }
+        $scope.formScope.showNotAllowedMsg = false;
+        return true;
+      };
+      $scope.isForm = true;
+      $scope.template = "demo-form.html";
+      $scope.formScope = {
+        inputs: {
+          first: "",
+          second: "test",
+          third: ""
+        },
+        allowBackgroundDismissal: false,
+        showNotAllowedMsg: false
+      };
+      $scope.actionButtons = [
+          {
+            label: "Cancel",
+            isCancel: true
+          },
+          {
+            label: "Save",
+            class: "btn-primary custom-class",
+            actionFn: function() {
+                $scope.actionsText = "inputs {" +
+                    "\n    first: " + $scope.formScope.inputs.first +
+                    "\n    second: " + $scope.formScope.inputs.second +
+                    "\n    third: " + $scope.formScope.inputs.third +
+                    "\n}" + $scope.actionsText;
+                return true;
```

Comment:
Sorry, my bad, my poor attempt to do post submission validation, which isn't the correct pattern.   actionFn's do not need to return T/F.